File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-shared-5-4/Frequency.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class Frequency : ISerialized
    {
        public Frequency(XmlNode xmlNode)
        {
            XmlNode periodMultiplierNode = xmlNode.SelectSingleNode("periodMultiplier");
            
            if (periodMultiplierNode != null)
            {
                if (periodMultiplierNode.Attributes["href"] != null || periodMultiplierNode.Attributes["id"] != null) 
                {
                    if (periodMultiplierNode.Attributes["id"] != null) 
                    {
                        periodMultiplierIDRef_ = periodMultiplierNode.Attributes["id"].Value;
                        XsdTypePositiveInteger ob = new XsdTypePositiveInteger(periodMultiplierNode);
                        IDManager.SetID(periodMultiplierIDRef_, ob);
                    }
                    else if (periodMultiplierNode.Attributes["href"] != null)
                    {
                        periodMultiplierIDRef_ = periodMultiplierNode.Attributes["href"].Value;
                    }
                    else
                    {
                        periodMultiplier_ = new XsdTypePositiveInteger(periodMultiplierNode);
                    }
                }
                else
                {
                    periodMultiplier_ = new XsdTypePositiveInteger(periodMultiplierNode);
                }
            }
            
        
            XmlNode periodNode = xmlNode.SelectSingleNode("period");
            
            if (periodNode != null)
            {
                if (periodNode.Attributes["href"] != null || periodNode.Attributes["id"] != null) 
                {
                    if (periodNode.Attributes["id"] != null) 
                    {
                        periodIDRef_ = periodNode.Attributes["id"].Value;
                        PeriodExtendedEnum ob = new PeriodExtendedEnum(periodNode);
                        IDManager.SetID(periodIDRef_, ob);
                    }
                    else if (periodNode.Attributes["href"] != null)
                    {
                        periodIDRef_ = periodNode.Attributes["href"].Value;
                    }
                    else
                    {
                        period_ = new PeriodExtendedEnum(periodNode);
                    }
                }
                else
                {
                    period_ = new PeriodExtendedEnum(periodNode);
                }
            }
            
        
        }
        
    
        #region PeriodMultiplier_
        private XsdTypePositiveInteger periodMultiplier_;
        public XsdTypePositiveInteger PeriodMultiplier_
        {
            get
            {
                if (this.periodMultiplier_ != null)
                {
                    return this.periodMultiplier_; 
                }
                else if (this.periodMultiplierIDRef_ != null)
                {
                    periodMultiplier_ = IDManager.getID(periodMultiplierIDRef_) as XsdTypePositiveInteger;
                    return this.periodMultiplier_; 
                }
                else
                {
                      return this.periodMultiplier_; 
                }
            }
            set
            {
                if (this.periodMultiplier_ != value)
                {
                    this.periodMultiplier_ = value;
                }
            }
        }
        #endregion
        
        public string periodMultiplierIDRef_ { get; set; }
        #region Period_
        private PeriodExtendedEnum period_;
        public PeriodExtendedEnum Period_
        {
            get
            {
                if (this.period_ != null)
                {
                    return this.period_; 
                }
                else if (this.periodIDRef_ != null)
                {
                    period_ = IDManager.getID(periodIDRef_) as PeriodExtendedEnum;
                    return this.period_; 
                }
                else
                {
                      return this.period_; 
                }
            }
            set
            {
                if (this.period_ != value)
                {
                    this.period_ = value;
                }
            }
        }
        #endregion
        
        public string periodIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-shared-5-4/Frequency.cpp
// Frequency.cpp 
#include "Frequency.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

Frequency::Frequency(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //periodMultiplierNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* periodMultiplierNode = xmlNode->FirstChildElement("periodMultiplier");

   if(periodMultiplierNode){periodMultiplierIsNull_ = false;}
   else{periodMultiplierIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- periodMultiplierNode , address : " << periodMultiplierNode << std::endl;
   #endif
   if(periodMultiplierNode)
   {
      if(periodMultiplierNode->Attribute("href") || periodMultiplierNode->Attribute("id"))
      {
          if(periodMultiplierNode->Attribute("id"))
          {
             periodMultiplierIDRef_ = periodMultiplierNode->Attribute("id");
             periodMultiplier_ = boost::shared_ptr<XsdTypePositiveInteger>(new XsdTypePositiveInteger(periodMultiplierNode));
             periodMultiplier_->setID(periodMultiplierIDRef_);
             IDManager::instance().setID(periodMultiplierIDRef_,periodMultiplier_);
          }
          else if(periodMultiplierNode->Attribute("href")) { periodMultiplierIDRef_ = periodMultiplierNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { periodMultiplier_ = boost::shared_ptr<XsdTypePositiveInteger>(new XsdTypePositiveInteger(periodMultiplierNode));}
   }

   //periodNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* periodNode = xmlNode->FirstChildElement("period");

   if(periodNode){periodIsNull_ = false;}
   else{periodIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- periodNode , address : " << periodNode << std::endl;
   #endif
   if(periodNode)
   {
      if(periodNode->Attribute("href") || periodNode->Attribute("id"))
      {
          if(periodNode->Attribute("id"))
          {
             periodIDRef_ = periodNode->Attribute("id");
             period_ = boost::shared_ptr<PeriodExtendedEnum>(new PeriodExtendedEnum(periodNode));
             period_->setID(periodIDRef_);
             IDManager::instance().setID(periodIDRef_,period_);
          }
          else if(periodNode->Attribute("href")) { periodIDRef_ = periodNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { period_ = boost::shared_ptr<PeriodExtendedEnum>(new PeriodExtendedEnum(periodNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypePositiveInteger> Frequency::getPeriodMultiplier()
{
   if(!this->periodMultiplierIsNull_){
        if(periodMultiplierIDRef_ != ""){
             return boost::shared_static_cast<XsdTypePositiveInteger>(IDManager::instance().getID(periodMultiplierIDRef_));
        }else{
             return this->periodMultiplier_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypePositiveInteger>();
   }
}
boost::shared_ptr<PeriodExtendedEnum> Frequency::getPeriod()
{
   if(!this->periodIsNull_){
        if(periodIDRef_ != ""){
             return boost::shared_static_cast<PeriodExtendedEnum>(IDManager::instance().getID(periodIDRef_));
        }else{
             return this->period_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PeriodExtendedEnum>();
   }
}
}

